### ISSUE TITLE
Reproduce the bone marrow cell classification (Table 3) in the paper

### ISSUE THREAD

**Issue by apple2373** [marrlab/DinoBloom#4]
I am trying to reproduce Table 3 in the paper "Evaluation on Bone Marrow: WBC Classification on the Dataset BMC with 21 Highly Imbalanced Classes." I believe I have found the exact data split as well as the evaluation script.

Data split: 
- Train: https://github.com/marrlab/DinoBloom/blob/7adf522bf3a07b990f5a151d4947e9bcf984dfc1/dinov2/eval/splits/bm_train.csv
- Test: https://github.com/marrlab/DinoBloom/blob/7adf522bf3a07b990f5a151d4947e9bcf984dfc1/dinov2/eval/splits/bm_test.csv
Evaluation script: 
- https://github.com/marrlab/DinoBloom/blob/7adf522bf3a07b990f5a151d4947e9bcf984dfc1/dinov2/eval/scripts/bm_eval.sh#L21
- https://github.com/marrlab/DinoBloom/blob/7adf522bf3a07b990f5a151d4947e9bcf984dfc1/dinov2/eval/general_fixed_split_patch_eval.py

After I fix path of images, removing wandb etc (and I attach [my modified version](https://github.com/user-attachments/files/19065615/test_pretrained_model.py.txt) in case), I was able to run the code using the pretrained weights provided in this code. I got mostly similar results but some metrics are significantly (more than 3 points) lower than reported. 

| Model                   | 1-NN wF1 | 1-NN Acc | 1-NN bAcc  | 20-NN wF1 | 20-NN Acc | 20-NN bAcc | Linear Probe wF1 | Linear Probe Acc | Linear Probe bAcc  |
|-------------------------|----------|----------|------------|-----------|-----------|------------|------------------|------------------|--------------------|
| DinoBloom-S (reported)  | 78.4     | 78.3     | 62.0       | 84.2      | 84.8      | 55.6       | 85.7             | 85.9             | 71.4               |
| DinoBloom-S (reproduced)| 77.2     | 77.1     | **56.6**   | 83.5      | 84.0      | 55.2       | 85.2             | 85.4             | **64.6**           |
| DinoBloom-B (reported)  | 79.6     | 79.5     | 65.8       | 83.7      | 84.1      | 57.1       | 85.5             | 85.6             | 70.7               |
| DinoBloom-B (reproduced)| 77.8     | 77.6     | **62.3**   | 83.6      | 84.0      | 55.6       | 85.1             | 85.2             | **64.9**           |
| DinoBloom-L (reported)  | 78.8     | 78.8     | 57.7       | 83.6      | 84.0      | 56.3       | 84.9             | 85.0             | 64.4               |
| DinoBloom-L (reproduced)| 78.8     | 78.7     | 57.9       | 83.6      | 84.1      | 56.3       | 85.0             | 85.1             | 64.2               |
| DinoBloom-G (reported)  | 80.0     | 79.9     | 59.4       | 83.8      | 84.2      | 56.2       | 84.9             | 85.0             | 69.3               |
| DinoBloom-G (reproduced)| 80.0     | 79.9     | 59.3       | 83.8      | 84.2      | 56.1       | 84.9             | 84.9             | 70.0               |

For example, the Linear Probe bAcc of DinoBloom-S is reported to be 71.4, but my reproduction yields 64.6. Is this expected? My speculation is that bAcc might have high variance because results from minor classes can heavily influence the score. However, I also wonder if using the exact same weights should yield identical results. Maybe, while the code and data split appear to match those used in the paper, it's possible that the weights are not exactly the same as those reported.

Any comments?

**Comment by ValentinKoch** [comment]
Hi, 
great effort, thanks for your interest in our work :)
I wouldnt expect this behaviour, no. I will try to look into it! 
Best,
Valentin

**Comment by apple2373** [comment]
Thanks for the reply. If possible, could you simply run general_fixed_split_patch_eval.py on your own environment using the DinoBloom-S weights provided in this repository, and tell me the metrics (eps bAcc) that you get? I want to know if my modification to fit for my environment decreased the accuracy or not.
